consent/handler.go

```diff
@@ -313,13 +313,22 @@ func (h *Handler) AcceptLoginRequest(w http.ResponseWriter, r *http.Request, ps
 		return
 	}
 
-	var p HandledLoginRequest
+	var (
+		p   HandledLoginRequest
+		err error
+	)
+
+	defer func(p *HandledLoginRequest, err error) {
+		recordAcceptLoginRequest(p, err)
+	}(&p, err)
```

Comment:
Relying on a global error is very flaky - there are several places where err will be overshadowed, e.g. [here](https://github.com/ory/hydra/pull/1381/files#diff-ac4ce4ab0ca6c69035745a50d3108c65R337) and [here](https://github.com/ory/hydra/pull/1381/files#diff-ac4ce4ab0ca6c69035745a50d3108c65R354) and [here](https://github.com/ory/hydra/pull/1381/files#diff-ac4ce4ab0ca6c69035745a50d3108c65R360). I'm not sure about other projects but I feel that multiple return values usually re-define/overshadow the original err scope:

```
package main

import (
	"fmt"
)

func main() {
	a := "a"
	fmt.Println(a) // "a"
	
	b, a := "b", "c"
	fmt.Println(a) // "c"
	fmt.Println(b) // "b"
}
```